src/PluginClientFactoryInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Http\Client\Common;
+
+use Http\Client\HttpAsyncClient;
+use Http\Client\HttpClient;
+
+/**
+ * Create a PluginClient.
+ *
+ * @author Fabien Bourigault <bourigaultfabien@gmail.com>
+ */
+interface PluginClientFactoryInterface
+{
+    /**
+     * @param HttpClient|HttpAsyncClient $client
+     * @param Plugin[]                   $plugins
+     * @param array                      $options
+     *
+     * @see PluginClient constructor for $options details.
```

Comment:
No, we cannot "see the PluginClient". We are not aware of any client implementation. We should write something like "array of options that are passed to the client."